src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -218,6 +220,25 @@ void QueryCache::Writer::finalizeWrite()
 
     std::lock_guard lock(mutex);
 
+    auto to_single_chunk = [](const std::shared_ptr<Chunks>& chunks_) -> std::shared_ptr<Chunks>
```

Comment:
If the resultset is large, making it a single chunk can be worse. Let's just squash to max_block_size.

Reply:
Yepp, will check.